tetris.py

```diff
@@ -80,17 +85,29 @@ def draw_block(self):
 
         self.canvas.delete("block")
         for x, y in (
-            self.tetris_game.get_current_block() + self.tetris_game.landed_blocks
+            self.tetris_game.get_current_block()
         ):
             self.canvas.create_rectangle(
                 x * square_size,
                 y * square_size,
                 x * square_size + square_size,
                 y * square_size + square_size,
                 tags="block",
-                fill=RED,
+                fill=color_dict[self.tetris_game.current_block_shape]
             )
 
+        for shape_letter, coords in self.tetris_game.landed_blocks.items():
+            color = color_dict[shape_letter]
+            for x, y in coords:
+                self.canvas.create_rectangle(
+                    x * square_size,
+                    y * square_size,
+                    x * square_size + square_size,
+                    y * square_size + square_size,
+                    tags="block",
+                    fill=color
+                )
```

Comment:
This is a bit copy/paste. If it bothers you, you can move the long `self.canvas.create_rectangle(...)` into a new function that takes `x`, `y` and `color` as arguments.